### PR TITLE
[MHA] turn on 'negative zero as nan' for FP8 numbers

### PR DESCRIPTION
**Pull request by CAHEK7**
Switch defaults to HW supported mode for soft-float calculations.
It does not affect GPU code, since it is already used the only available HW mode, but it affects all the host calculations like CPU verification code.

Related to #2971 and blocked by #3001
Probably related to #2996

### CODE REVIEW

**Comment by junliume** [comment]
@CAHEK7 @atamazov so we need to regenerate KDB caches after this change too?

***update***: `Dbsync gfx90a` test indeed is failing with this PR

**Comment by CAHEK7** [comment]
After exactly this branch.
I reverted this change in my previous branch and disabled my fp8 test if that macro is 1.
It doesn't block fp8 mha, since GPU code has hw compatible implementation, but any host-based internal miopen fp8 calculations won't be correct, that's why I disabled the test.

**UPD**: the main problem is that the macro goes to all the kernels, even if they do not use it and do not support FP8 at all. It invalidates almost everything even on the targets without FP8.

**Comment by junliume** [comment]
@CAHEK7 and @cderb will `gfx942`'s KDB have the same issues after this PR? We cannot merge it before confirming.

**Comment by cderb** [comment]
> @CAHEK7 and @cderb will `gfx942`'s KDB have the same issues after this PR? We cannot merge it before confirming.

gfx942 dbsync is passing when I run it by hand.

**Review by junliume — APPROVED** [review]

**Comment by atamazov** [comment]
@CAHEK7 @JehandadKhan 

> **UPD**: the main problem is that the macro goes to all the kernels, even if they do not use it and do not support FP8 at all. It invalidates almost everything even on the targets without FP8.

I think that this "selection of F8 type" shall be completely removed in the future. We shall switch to _distinct_ F8 types. That would automatically get rid of all this mess.

[Informative] Right now I foresee three F8 types:
- The F8 type which works on current HW (`bf8_fnuz`)
- Two F8 types that comply the OFP8 spec (FP8 in E4M3 format and BF8 in E5M2).

**Pull request merged**
